esm_master/esm_master.py

```diff
@@ -32,21 +36,38 @@ def main_flow(parsed_args, target):
     # Miguel: Move this somewhere else after talking to Paul and Dirk
     user_config["general"]["run_or_compile"] = "compiletime"
 
+    # deniz: small bugfix: when esm_master receives --verbose, it did not make
+    # it into the configuration since it was only a global variable
+    if verbose:
+        user_config["general"]["verbose"] = True
+    
     from esm_runscripts.sim_objects import SimulationSetup
     complete_setup = SimulationSetup(user_config=user_config)
     complete_config = complete_setup.config
 
     # This will be a problem later with GEOMAR
     #setups2models.replace_last_vars(env)
 
+
     user_task = Task(target, setups2models, vcs, main_infos, complete_config)
+
     if verbose > 0:
         user_task.output()
 
     user_task.output_steps()
 
     if check:
+        # deniz: if the environment variable ESM_MASTER_DEBUG is also set dump
+        # the contents of the current config to stdout for more investigation 
+        if os.environ.get("ESM_MASTER_DEBUG", None):
```

Comment:
Hi Deniz, for info:

```python
d = {"a": 1, "b": 2, "c": 3}
answer = d.get("d")
assert answer is None 
```

Notice that this doesn't raise an `AssertionError`. The second argument to `d.get` is the value that should be returned if the key is not in the dictionary. That defaults to `None`.

Reply:
That's right. I think it is cleaner to use `os.getenv()` instead which is actually a wrapper around `os.environ.get()`:
```python
import os
os.getenv??

Signature: os.getenv(key, default=None)
Source:
def getenv(key, default=None):
    """Get an environment variable, return None if it doesn't exist.
    The optional second argument can specify an alternate default.
    key, default and the result are str."""
    return environ.get(key, default)
File:      /sw/rhel6-x64/conda/anaconda3-bleeding_edge/lib/python3.6/os.py
Type:      function
```